Makefile

```diff
@@ -8,11 +10,21 @@ build:
 format:
 	find . -name '*.go' -exec gofmt -s -w {} +
 
-.PHONY: check
-check:
+.PHONY: ci-like-lint
+ci-like-lint :
+	docker run --rm -v $(shell pwd):/app -w /app golangci/golangci-lint:$(GOLANGCI_VERSION) make lint
```

Comment:
I can comment on it, but the main difference is that the `make lint` runs the `golangci-lint` on the host machine, which is sometimes not the same as in the CI. That's fine for most of the time because it's speedy. 

But sometimes, the analysts between the versions are different (let say the new version has new default linters). And for such cases, the `make ci-like-lint` can be used. It will run linter using the same version as in the CI.

Reply:
Well, that's true that the difference is not so important, but it's still significant. If you try, you can easily spot it.

The other thing is that both targets are unique and do not duplicate each other since the `make ci-like-lint` is executing the `make lint`.

For example, on my local machine (Linux), even if the image is loaded, it takes ~30 seconds to run the container, do the job, and clean it up (for sure, the image is loaded). If the linter is running on a host machine, it is under 2 secs.

Anyway, can I wonder what the reasons behind the change request are? I mean sorry, but usually, I need some elaborations for the statements that started like "I prefer" :smiley:

Reply:
>For example, on my local machine (Linux), even if the image is loaded, it takes ~30 seconds to run the container, do the job, and clean it up (for sure, the image is loaded). If the linter is running on a host machine, it is under 2 secs.

On my machine, I see under 5s duration only after the first time (~15s) so I expect the cache is impacting the results, and I don't know how much is real in daily work to have the cached build, in my experience the time is most close to the first time, but I don't really have good data to make it scientific.

>Anyway, can I wonder what the reasons behind the change request are?

Mainly why docker exists, so to avoid the dependency from the underhood system. So the team member is sure to run commands with the same dependencies and versions and external contributors shouldn't care about them (making the contribution process easier).

Reply:
>Mainly why docker exists, so to avoid the dependency from the underhood system.

I fully agree with that, and basically, I was following this when I introduced that target in our Makefile. Moreover, it can be the default part of the `make check` by the nature of this target. But I have a feeling that we should hear more opinions on that.

The only thing that I disagree with is that we should keep only the single target for running the linter (basically remove the hosted) because I see a benefit of having freedom of choosing what to run and see no issues with having this freedom :man_shrugging: Maybe I'm missing something :thinking:

Reply:
I'd personally benefit from `ci-like-lint`, since I often have a different version of golangci-lint on my machine. To improve the performance of the docker version, maybe we can create a `/tmp/golangci-cache-${GOLANGCI_LINT_VERSION}` folder, pass it as a docker volume and use https://golangci-lint.run/usage/configuration/#cache ?

Reply:
I don't really want to start having some random places where there are additional caches, also it seems like this will not work on windows ;) Although I don't know how docker on windows works, I guess wsl(2) can have `/tmp` :thinking:

Reply:
We can use something like this to be extra safe:
```
"${TMPDIR:-/tmp}/golangci-cache-${GOLANGCI_LINT_VERSION:-all}"
```